src/graph_building/edge_finder.py

```diff
@@ -1,6 +1,76 @@
+from itertools import product
+
+import pandas as pd
+
+from src.graph_building.interfaces.edge_interface import EdgeInterface
+from src.graph_building.interfaces.vertex_data_interface import VertexDataInterface
+
+
 class EdgeFinder:
     """
     This class is responsible for finding edges between vertices.
     """
-    def __init__(self):
-        pass
+    def __init__(self,
+                 gauges: list,
+                 beta: int
+                 ):
+        """
+        Constructor.
+        :param list gauges: list of stations
+        :param int beta: the number of days allowed after a vertex for continuation
+        """
+        self.gauges = gauges
+        self.beta = beta
+
+        self.edge_interface = EdgeInterface()
+
+    def run(self, vertex_interface: VertexDataInterface):
+        """
+        We take neighboring gauges and find all edges going between them.
+        :param VertexDataInterface vertex_interface: interface with vertices
+        """
+        edges = dict()
+        for upstream, downstream in zip(self.gauges[:-1], self.gauges[1:]):
+            edges[f"{upstream}-{downstream}"] = self.find_edges(
+                upstream_vertices=vertex_interface.vertices[upstream],
```

Comment:
A két átadott értékre (LOC 35-36) külön változót vezetnék be ugyanazzal a névvel, ami az argumentumnév is.